lib/extep.ex

```diff
@@ -182,16 +189,20 @@ defmodule Extep do
       iex> Extep.new(%{user_id: 1})
       ...> |> Extep.async(:user, fn ctx -> {:ok, %{id: ctx.user_id, name: "Alice"}} end)
       ...> |> Extep.async(:items, fn _ctx -> {:ok, [%{code: "item1"}]} end)
-      ...> |> Extep.await()
+      ...> |> Extep.await(timeout: 10)
       %Extep{status: :ok, context: %{user_id: 1, user: %{id: 1, name: "Alice"}, items: [%{code: "item1"}]}, message: nil}
 
   Note: `await/1` is automatically called by `run/2`, `run/3`, and `return/2` when there are pending tasks.
   """
-  @spec await(t()) :: t()
-  def await(%{tasks: [_ | _]} = extep) when is_ok(extep) do
+  @spec await(%{tasks: [Task.t()], context: map()}, keyword()) :: map()
```

Comment:
I think we can keep `t()` as the type for the first arg, WDYT?
```suggestion
  @spec await(t(), keyword()) :: map()
```

Reply:
Done ✅